### PR TITLE
Update node density docs

### LINKED ISSUE #7443: Update node density docs
With @ajwerner 's larger range size change and testing ([report](https://docs.google.com/document/d/19ystASmq_14B9zzzsn0tQTu6LCahwh6bFyBfqjHEfS8/edit#))in 20.1, we can handle more storage per node. We should update our [production checklist docs](https://www.cockroachlabs.com/docs/stable/recommended-production-settings.html) to maintain one place for this type of info. Let's document four things:

1. Range size went from 64MiB to 512MiB in 20.1.
2. Users should do their own testing to arrive at their optimal storage per node because it varies by workload.
3. With our own internal testing using the [bank workload](https://www.cockroachlabs.com/docs/stable/cockroach-workload.html#bank-workload), we arrived at successfully storing 4.32 TiB logical data per node. 
4. Configuration and full results of our applying bank workload:

- 6x c5d.4xlarge nodes, gp2 EBS with 5TiB each

Results (could present it in a table):

- vCPU Per Node: 16

- Logical Data Per Node: 4.32 TiB

- Ram Per Node: 32 GiB

- Data Per Core: ~270 GiB / vCPU

-  Data Per Ram: ~135 GiB / GiB

### PR DESCRIPTION
**Pull request by rmloveland**
Fixes #7443.

### CODE REVIEW

**Comment by rmloveland** [comment]
@ricardocrdb this is the "max node size" docs PR we discussed.  Please take a look and let me know what you think.

(PS random aside: I tried adding you as a reviewer but the Github UI could not complete your GH username for some reason.)

**Comment by johnrk** [comment]
looks good to me

File: v20.1/recommended-production-settings.md [hunk 1]
```diff
@@ -47,9 +47,13 @@ Nodes should have sufficient CPU, RAM, network, and storage capacity to handle y
 
 - Avoid using shared storage such as NFS, CIFS, and CEPH storage.
 
-- For the best performance results, use SSD or NVMe devices. The recommended volume size is 300 GB - 2 TB.
+- For the best performance results, use SSD or NVMe devices. The recommended volume size is 300 GiB - 2 TiB.
 
-    Monitor IOPS for higher service times. If they exceed 1-5 ms, you will need to add more devices or expand the cluster to reduce the disk latency. To monitor IOPS, use tools such as `iostat` (part of `sysstat`).
+    {{site.data.alerts.callout_info}}
+    Depending on the workload, you may be able to store more than 2 TiB per node. In a narrowly-scoped test, we were able to successfully store 4.32 TiB of logical data per node. The results of this test may not be applicable to your specific situation; testing with your workload is _strongly_ recommended. For more information about this test, see [Node density testing configuration](#node-density-testing-configuration).
```
Comment (ricardocrdb) [inline v20.1/recommended-production-settings.md:53]: Suggestion:
> testing with your workload is _strongly_ recommended.

and changing:
> testing with your workload is _strongly_ recommended before using it in a production environment.

File: v20.1/recommended-production-settings.md [hunk 2]
```diff
@@ -61,6 +65,22 @@ Nodes should have sufficient CPU, RAM, network, and storage capacity to handle y
 
 - The optimal configuration for striping more than one device is [RAID 10](https://en.wikipedia.org/wiki/Nested_RAID_levels#RAID_10_(RAID_1+0)). RAID 0 and 1 are also acceptable from a performance perspective.
 
+##### Node density testing configuration
+
+In a narrowly-scoped test, we were able to successfully store 4.32 TiB of logical data per node. The results of this test may not be applicable to your specific situation; testing with your workload is _strongly_ recommended.
```
Comment (ricardocrdb) [inline v20.1/recommended-production-settings.md:70]: Suggestion is: 
> testing with your workload is _strongly_ recommended.

and changing:
> testing with your workload is _strongly_ recommended before using it in a production environment.

File: v20.2/recommended-production-settings.md [hunk 3]
```diff
@@ -47,9 +47,13 @@ Nodes should have sufficient CPU, RAM, network, and storage capacity to handle y
 
 - Avoid using shared storage such as NFS, CIFS, and CEPH storage.
 
-- For the best performance results, use SSD or NVMe devices. The recommended volume size is 300 GB - 2 TB.
+- For the best performance results, use SSD or NVMe devices. The recommended volume size is 300 GiB - 2 TiB.
 
-    Monitor IOPS for higher service times. If they exceed 1-5 ms, you will need to add more devices or expand the cluster to reduce the disk latency. To monitor IOPS, use tools such as `iostat` (part of `sysstat`).
+    {{site.data.alerts.callout_info}}
+    Depending on the workload, you may be able to store more than 2 TiB per node. In a narrowly-scoped test, we were able to successfully store 4.32 TiB of logical data per node. The results of this test may not be applicable to your specific situation; testing with your workload is _strongly_ recommended. For more information about this test, see [Node density testing configuration](#node-density-testing-configuration).
```
Comment (ricardocrdb) [inline v20.2/recommended-production-settings.md:53]: Suggestion:
> testing with your workload is _strongly_ recommended.

and changing:
> testing with your workload is _strongly_ recommended before using it in a production environment.

File: v20.2/recommended-production-settings.md [hunk 4]
```diff
@@ -61,6 +65,22 @@ Nodes should have sufficient CPU, RAM, network, and storage capacity to handle y
 
 - The optimal configuration for striping more than one device is [RAID 10](https://en.wikipedia.org/wiki/Nested_RAID_levels#RAID_10_(RAID_1+0)). RAID 0 and 1 are also acceptable from a performance perspective.
 
+##### Node density testing configuration
+
+In a narrowly-scoped test, we were able to successfully store 4.32 TiB of logical data per node. The results of this test may not be applicable to your specific situation; testing with your workload is _strongly_ recommended.
```
Comment (ricardocrdb) [inline v20.2/recommended-production-settings.md:70]: Suggestion:
> testing with your workload is _strongly_ recommended.

and changing:
> testing with your workload is _strongly_ recommended before using it in a production environment.

File: v20.1/recommended-production-settings.md (hunk 1 above)
Comment (rmloveland) [reply v20.1/recommended-production-settings.md:53]: Done.

File: v20.1/recommended-production-settings.md (hunk 2 above)
Comment (rmloveland) [reply v20.1/recommended-production-settings.md:70]: Done.

File: v20.2/recommended-production-settings.md (hunk 3 above)
Comment (rmloveland) [reply v20.2/recommended-production-settings.md:53]: Done.

File: v20.2/recommended-production-settings.md (hunk 4 above)
Comment (rmloveland) [reply v20.2/recommended-production-settings.md:70]: Done.

**Comment by rmloveland** [comment]
Thanks Ricardo! I am going to merge this with your suggested updates.

Thanks John and Andrew as well!

**Pull request merged**